src/object/obj_class.c

```diff
@@ -211,44 +211,63 @@ dc_set_oclass(daos_handle_t coh, int domain_nr, int target_nr,
 	/** first set a reasonable default based on RF & RDD hint (if set) */
 	switch (rf_factor) {
 	case DAOS_PROP_CO_REDUN_RF0:
-		if (rdd == DAOS_OCH_RDD_RP)
+		if (rdd == DAOS_OCH_RDD_RP) {
 			cid = OC_RP_2GX;
-		else if (rdd == DAOS_OCH_RDD_EC)
-			cid = OC_EC_2P1G1;
-		else
+		} else if (rdd == DAOS_OCH_RDD_EC) {
+			if (domain_nr >= 10)
+				cid = OC_EC_8P1GX;
+			else if (domain_nr >= 6)
+				cid = OC_EC_4P1GX;
+			else
+				cid = OC_EC_2P1GX;
```

Comment:
probably EC should not be allowed for RF0, i.e. only striping is allowed here.

Reply:
i don't see why one can't use EC with RF0.. in my understanding, that with RF0 redundancy is not required but one can still do repl or EC; maybe i misunderstood?.
and here we set to EC only if the EC hint is explicitly provided by the user..

Reply:
oh, yes, you are right, this is not required. Sorry for noise.